enonic/src/main/resources/site/parts/comments-list/comments-list.ts

```diff
@@ -0,0 +1,230 @@
+import {Content, QueryResponse} from "enonic-types/lib/content";
+import {sequenceT} from "fp-ts/lib/Apply";
+import * as A from "fp-ts/lib/Array";
+import {query} from "enonic-fp/lib/content";
+import {flow, tupled} from "fp-ts/lib/function";
+import {Comment} from "../../content-types/comment/comment";
+import {findUsers, getUser} from "enonic-fp/lib/auth";
+import {User} from "enonic-types/lib/context";
+import {EnonicError} from "enonic-fp/lib/errors";
+import {contramap, ordBoolean} from "fp-ts/lib/Ord";
+import {isSome, Option} from "fp-ts/lib/Option";
+import {CommentsListPartConfig} from "./comments-list-part-config";
+import {Component} from "enonic-types/lib/portal";
+import {formatWithOptions, parseISO,} from "date-fns/fp";
+
+type CommentsListPart = Component<CommentsListPartConfig>
+export type UserMapping = Record<string, string>;
+
+const view = resolve('./comments-list.html')
+const renderer = getRenderer<ThymeleafParams>(view);
+
+
+export function get(req: Request): Response {
+  return pipe(
+    sequenceT(ioEither)(
+      getComponent<CommentsListPartConfig>(),
+      getContent(),
+    ),
+    chain(tupled((part: CommentsListPart, content: Content) => {
+      return sequenceT(ioEither)(
+        query<Comment>({
+          count: 100,
+          query: `_parentPath LIKE '/content${content._path}'`,
+          contentTypes: [`${app.name}:comment`]
+        }),
+        right(part),
+        right(content._path)
+      )
+    })),
+      chain(tupled((
+        resComments: QueryResponse<Comment>,
+        part: CommentsListPart,
+        parentPath: string,
+      ) => {
+        return sequenceT(ioEither)(
+          getUserMapping(resComments.hits),
+          right(resComments.hits),
+          right(getUser()),
+          right(part),
+          right(parentPath),
+        )
+      })),
+      chain(tupled((
+        userMapping: UserMapping,
+        comments: ReadonlyArray<Content<Comment>>,
+        user: Option<User>,
+        part: CommentsListPart,
+        parentPath: string,
+      ) => {
+        return sequenceT(ioEither)(
+          renderer(createThymeleafParams(userMapping, comments, isSome(user), req.path, part.config,  parentPath)),
+          right(part.config.includeCss === 'true')
+        )
+      })),
+      fold(
+        errorResponse('commentList', true),
+        tupled((res: string, includeCss: boolean) => ok(res, includePageContributions(includeCss)))
+      )
+    )();
+}
+
+function cssLink() {
+  return `<link
+           rel="stylesheet"
+           href="${assetUrl({path: 'design-system/bundle.css'})}" />`
+}
+
+function includePageContributions (includeCss: boolean) {
+  return includeCss ? {
+    pageContributions: {
+      headEnd: cssLink()
+    }
+  } : undefined
+}
+
+export function getUserMapping(res: ReadonlyArray<Content<Comment>>): IOEither<EnonicError, UserMapping> {
+
+  const set = res.map((c) => `"${c.creator}"`).join(',')
```

Comment:
I don't think the variable names is helping the reader enough here. The reader of the code should not have to interpret single letter names such as `c`.

 Aim for clarity when you are naming, and  don't be afraid to be verbose.

These are my suggestions:

```suggestion
  const creatorNames = comments
    .map((comment) => `"${comment.creator}"`)
    .join(',')
```